src/OpenTelemetry.Exporter.Geneva/GenevaLogExporter.cs

```diff
@@ -443,5 +470,70 @@ private static byte GetSeverityNumber(LogLevel logLevel)
                 return 1;
         }
     }
+
+    private static string Sanitize(string categoryName)
+    {
+        Span<char> tempArrSpan = stackalloc char[200];
+        int readIdx = 0;
+        int writeIdx = 0;
+
+        if (categoryName.Length > 0)
+        {
+            var firstChar = categoryName[0]; // special treatment for the first character.
+            if (firstChar >= 'A' && firstChar <= 'Z')
+            {
+                tempArrSpan[writeIdx] = firstChar;
+                ++writeIdx;
+            }
+            else if (firstChar >= 'a' && firstChar <= 'z')
+            {
+                // If the first character in the resulting string is lower-case ALPHA,
+                // it will be converted to the corresponding upper-case.
+                tempArrSpan[writeIdx] = (char)(firstChar - 32);
+                ++writeIdx;
+            }
+
+            // Not a valid name - Part B name should follow PascalCase naming convention.
+            else
+            {
+                return null;
+            }
```

Comment:
nit: Putting the comment outside of the `else` here makes this unusual/hard to read, IMO. Suggest doing it like this:

```csharp
            }
            else
            {
                // Not a valid name - Part B name should follow PascalCase naming convention.
                return null;
            }
```

Reply:
I'm with you! I've adopted this pattern for comments in the new PR here: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/345
Thank you.